internal/storage/sql/base.go

```diff
@@ -41,6 +41,38 @@ func NewBaseStorage(db *sql.DB, dialect SQLDialect, tableName string) *BaseStora
 
 // StoreEvent stores a webhook event in the database
 func (s *BaseStorage) StoreEvent(ctx context.Context, event *storage.Event) error {
+	// Check if event exists
+	exists := false
+	err := s.builder.Select("1").
+		From(s.tableName).
+		Where("id = ?", event.ID).
+		RunWith(s.db).
+		QueryRowContext(ctx).
+		Scan(&exists)
+	if err != nil && err != sql.ErrNoRows {
+		return fmt.Errorf("checking event existence: %w", err)
+	}
+
+	if exists {
```

Comment:
In what cases do we think we'll see this?

Reply:
Good question. I don't know. I'm trying to figure out what other context lead it to think it might be required to port over.

Reply:
I think we should just drop events that already exist?